consensus/blake3pow/consensus.go

```diff
@@ -491,7 +491,13 @@ func (blake3pow *Blake3pow) CalcDifficulty(chain consensus.ChainHeaderReader, pa
 
 	parentOfParent := chain.GetHeaderByHash(parent.ParentHash())
 	if parentOfParent == nil || chain.IsGenesisHash(parentOfParent.Hash()) {
-		return parent.Difficulty()
+		numRegions, numZones := common.GetHierarchySizeForExpansionNumber(chain.GetExpansionNumber()) // Is the new expansion number correctly set before this?
+		numChains := new(big.Int).Mul(new(big.Int).SetUint64(numRegions), new(big.Int).SetUint64(numZones))
+		newDifficulty := new(big.Int).Div(parent.Difficulty(), numChains)
```

Comment:
this is incorrect. New difficulty should be _global_ difficulty divided by number of new chains. You are currently dividing the _local_ difficulty.

Global difficulty is collected in the prime header. Every new chain should sample the global difficulty at the prime "activation" block.